ops/framework.py

```diff
@@ -853,34 +854,33 @@ def reemit(self):
         """
         self._reemit()
 
-    def _reemit(self, single_event_path: str = None):
-
-        class EventContext:
-            """Handles toggling the hook-is-running state in backends.
+    @contextmanager
+    def event_context(self, event_name: str):
```

Comment:
Looks like this needs to be private.

Reply:
Good point, we don't want to risk people touching it from charm code.